ports/qemu-riscv/Makefile

```diff
@@ -0,0 +1,129 @@
+include ../../py/mkenv.mk
+-include mpconfigport.mk
+
+# qstr definitions (must come before including py.mk)
+QSTR_DEFS = qstrdefsport.h
+
+# MicroPython feature configurations
+MICROPY_ROM_TEXT_COMPRESSION ?= 1
+
+# include py core make definitions
+include $(TOP)/py/py.mk
+include $(TOP)/extmod/extmod.mk
+
+BOARD ?= virt
+
+CROSS_COMPILE ?= riscv64-unknown-elf-
+
+GCC_VERSION = $(word 1, $(subst ., , $(shell $(CC) -dumpversion)))
+
+# Ubuntu 22.04 ships with Picolibc rather than newlib.  To avoid having
+# per-distro workarounds, and given that Picolibc may get more support
+# in the future, it is chosen over newlib if it is found.
+PICOLIBC_SPECS = $(shell $(CC) --print-file-name=picolibc.specs)
+ifeq ($(PICOLIBC_SPECS),picolibc.specs)
+# Picolibc was not found.
+SPECS_FRAGMENT =
+else
+SPECS_FRAGMENT = --specs=$(PICOLIBC_SPECS)
```

Comment:
What do you get on Ubuntu 22.04 if you don't pass this --specs argument?

Reply:
I believe we've been through this before: https://github.com/micropython/micropython/pull/12853#discussion_r1593895204 :)  In short, no libc is picked up at all and compilation fails.

Reply:
Derp, yes, sorry for the duplicate.

For the next time I inevitably forget again, would it be OK to update the comment a little? Suggest something like:

```
# If Picolibc is available then select it explicitly. Ubuntu 22.04 ships riscv toolchain
# with Picolibc rather than Newlib, and the default is "nosys" so a value must be provided.
# To avoid having per-distro workarounds, always select Picolibc if available.
```

Reply:
A slightly edited version of your comment is in the makefile (a clarification on which RISC-V toolchain uses Picolibc, Ubuntu also ships a RV64/ELF toolchain).